### ISSUE TITLE
object compute.banksymatrics not found 

### ISSUE THREAD

**Issue by alirezaah98** [prabhakarlab/Banksy#17]
hello, 
I have been trying to use Banksy on Xenium data and I have also tried to embed the coordinates inside the object but I still can't runBanksy function on ir because it lacks computebanksy matrics and apparently that function is not being run on Seurat object, so I was wondering if you could help me with it .
Thanks

**Comment by jleechung** [comment]
Hi @alirezaah98, thanks for your interest in BANKSY and raising this issue. Could you please update your packages to the latest versions?

Install the `feat-aft` branch of our SeuratWrappers fork:
```
remotes::install_github('jleechung/seurat-wrappers@feat-aft')
```
Install the latest BANKSY version:
```
remotes::install_github('prabhakarlab/Banksy')
```

**Comment by alirezaah98** [comment]
But still the problem persists even after updating the packages 
Fetching data from slot data from assay RNA
Computing neighbor matrix
Error in RunBanksy(wt_seurat, lambda = 0.2, dimx = "x", dimy = "x", assay = "Xenium",  : 
  object 'compute.banksyMatrices' not found

**Comment by jleechung** [comment]
Can you provide a minimal reproducible example, and print session information `sessionInfo()`?

**Comment by alirezaah98** [comment]
Tried with both but still not working 
library(Banksy)
library(Seurat)
library(SeuratData)
library(SeuratWrappers)

library(ggplot2)
library(gridExtra)
library(pals)

# Kelly palette for visualization
mypal <- kelly()[-1]

wt <- readRDS('/slide1/slide1_g896.rds')


wt <- RunBanksy(wt, lambda = 0.2, verbose=TRUE, 
                      assay = 'Xenium', slot = 'data', features = 'all',
                      k_geom = 15)

query.counts <- GetAssayData(wt, assay = "Xenium", layer  = "counts")[, Cells(wt[["fov"]])]
coords <- GetTissueCoordinates(wt[["fov"]], which = "centroids")


wt_seurat <- CreateSeuratObject(counts = query.counts , meta.data = coords)


wt_seurat <- NormalizeData(wt_seurat, scale.factor = 100, normalization.method = 'RC')
wt_seurat<- ScaleData(wt_seurat)


# Run BANKSY
wt_seurat <- RunBanksy(wt_seurat, lambda = 0.2, dimx = 'x', dimy = 'x', 
                      assay = 'RNA', slot = 'data', features = 'all', k_geom = 10)


R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.9 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblas-r0.3.15.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Amsterdam
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Rfast_2.0.8           RcppZiggurat_0.1.6    Rcpp_1.0.12           BiocGenerics_0.48.1  
 [5] MatrixGenerics_1.14.0 matrixStats_1.2.0     dplyr_1.1.4           spacexr_2.2.1        
 [9] future_1.33.1         pals_1.8              gridExtra_2.3         ggplot2_3.5.0        
[13] SeuratWrappers_0.3.4  SeuratData_0.2.2.9001 Seurat_5.0.2          SeuratObject_5.0.1   
[17] sp_2.1-3              Banksy_0.99.7        

loaded via a namespace (and not attached):
  [1] RcppHungarian_0.3           RcppAnnoy_0.0.22            splines_4.3.1              
  [4] later_1.3.2                 bitops_1.0-7                tibble_3.2.1               
  [7] R.oo_1.26.0                 polyclip_1.10-6             fastDummies_1.7.3          
 [10] lifecycle_1.0.4             aricode_1.0.3               doParallel_1.0.17          
 [13] processx_3.8.3              globals_0.16.2              lattice_0.22-5             
 [16] MASS_7.3-60.0.1             magrittr_2.0.3              plotly_4.10.4              
 [19] remotes_2.4.2.1             httpuv_1.6.14               sctransform_0.4.1          
 [22] spam_2.10-0                 sessioninfo_1.2.2           pkgbuild_1.4.3             
 [25] spatstat.sparse_3.0-3       reticulate_1.35.0           cowplot_1.1.3              
 [28] mapproj_1.2.11              pbapply_1.7-2               RColorBrewer_1.1-3         
 [31] pkgload_1.3.4               maps_3.4.2                  abind_1.4-5                
 [34] zlibbioc_1.48.0             quadprog_1.5-8              Rtsne_0.17                 
 [37] GenomicRanges_1.54.1        purrr_1.0.2                 R.utils_2.12.3             
 [40] RCurl_1.98-1.14             rappdirs_0.3.3              GenomeInfoDbData_1.2.11    
 [43] IRanges_2.36.0              S4Vectors_0.40.2            ggrepel_0.9.5              
 [46] irlba_2.3.5.1               listenv_0.9.1               spatstat.utils_3.0-4       
 [49] goftest_1.2-3               RSpectra_0.16-1             spatstat.random_3.2-2      
 [52] fitdistrplus_1.1-11         parallelly_1.37.1           leiden_0.4.3.1             
 [55] codetools_0.2-19            DelayedArray_0.28.0         tidyselect_1.2.0           
 [58] spatstat.explore_3.2-6      jsonlite_1.8.8              ellipsis_0.3.2             
 [61] progressr_0.14.0            ggridges_0.5.6              survival_3.5-8             
 [64] iterators_1.0.14            foreach_1.5.2               dbscan_1.1-12              
 [67] tools_4.3.1                 ica_1.0-3                   glue_1.7.0                 
 [70] usethis_2.2.3               GenomeInfoDb_1.38.6         withr_3.0.0                
 [73] BiocManager_1.30.22         fastmap_1.1.1               fansi_1.0.6                
 [76] callr_3.7.5                 digest_0.6.34               rsvd_1.0.5                 
 [79] R6_2.5.1                    mime_0.12                   colorspace_2.1-0           
 [82] scattermore_1.2             sccore_1.0.5                tensor_1.5                 
 [85] dichromat_2.0-0.1           spatstat.data_3.0-4         R.methodsS3_1.8.2          
 [88] utf8_1.2.4                  tidyr_1.3.1                 generics_0.1.3             
 [91] data.table_1.15.2           httr_1.4.7                  htmlwidgets_1.6.4          
 [94] uwot_0.1.16                 pkgconfig_2.0.3             gtable_0.3.4               
 [97] lmtest_0.9-40               SingleCellExperiment_1.24.0 XVector_0.42.0             
[100] htmltools_0.5.7             profvis_0.3.8               dotCall64_1.1-1            
[103] scales_1.3.0                Biobase_2.62.0              png_0.1-8                  
[106] SpatialExperiment_1.12.0    rstudioapi_0.15.0           reshape2_1.4.4             
[109] rjson_0.2.21                curl_5.2.0                  nlme_3.1-164               
[112] cachem_1.0.8                zoo_1.8-12                  stringr_1.5.1              
[115] KernSmooth_2.23-22          parallel_4.3.1              miniUI_0.1.1.1             
[118] desc_1.4.3                  pillar_1.9.0                grid_4.3.1                 
[121] vctrs_0.6.5                 RANN_2.6.1                  urlchecker_1.0.1           
[124] promises_1.2.1              xtable_1.8-4                cluster_2.1.6              
[127] magick_2.8.3                cli_3.6.2                   compiler_4.3.1             
[130] rlang_1.1.3                 crayon_1.5.2                future.apply_1.11.1        
[133] mclust_6.1                  ps_1.7.6                    fs_1.6.3                   
[136] plyr_1.8.9                  stringi_1.8.3               viridisLite_0.4.2          
[139] deldir_2.0-4                munsell_0.5.0               lazyeval_0.2.2             
[142] devtools_2.4.5              spatstat.geom_3.2-9         Matrix_1.6-5               
[145] RcppHNSW_0.6.0              patchwork_1.2.0             shiny_1.8.0                
[148] SummarizedExperiment_1.32.0 ROCR_1.0-11                 memoise_2.0.1              
[151] leidenAlg_1.1.3             igraph_2.0.2

**Comment by alirezaah98** [comment]
Do I have to convert the Seurat object to Spatialexperiment object in order to run the algorithm?

**Comment by jleechung** [comment]
You can also do that, and follow the [vignette here](https://github.com/prabhakarlab/Banksy?tab=readme-ov-file#quick-start). Though, I'm curious why `SeuratWrappers::RunBanksy` doesn't work - the `compute.banksyMatrices` function should not be called. If you're willing to share your data, I'll be happy to troubleshoot it for you.

**Comment by alirezaah98** [comment]
yeah , problem solved , perhaps just a bug in Rstudio, Thank you for your help
